src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -783,7 +783,42 @@ internal enum GCConfigurationType
         [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "GCInterface_EnumerateConfigurationValues")]
         internal static unsafe partial void _EnumerateConfigurationValues(void* configurationDictionary, delegate* unmanaged<void*, void*, void*, GCConfigurationType, long, void> callback);
 
-        private static int _RefreshMemoryLimit()
+        internal enum RefreshMemoryStatus
+        {
+            Succeeded = 0,
+            HardLimitTooLow = 1,
+            HardLimitInvalid = 2,
+        }
+
+        /// <summary>
+        ///
+        /// Instructs the Garbage Collector to reconfigure itself by detecting the various memory limits on the system.
```

Comment:
probably want to say something like, this API will only handle configs that could be handled when the runtime is loaded, for example, for configs that don't have any effects on 32-bit systems (like the GCHeapHardLimit* ones), calling this API will not change that, ie, they would still not have any effect. but for configs like GCHighMemPercent, we will handle the new value you specify via AppContext